README.md

```diff
@@ -1,63 +1,88 @@
+```jinja
+<steppedValueSet variable="Qin_average" first="30" step="{{Qin_average_step}}" last="200"/>
 ```
 
-With the data file:
+In this example we converted the hard-coded value of `10` into a variable called `Qin_average_step` which will be used to generate a new code. The values this variable can take will be defined in the next step.
+
+### Step 2: Define the variables
+
+Now we have a variable called `Qin_average_step` in our example file (`experiments.xml`) and we want to generate different versions of it. In each version we we will replace `Qin_average_step` with a value we want.
+
+The values which the jinja templates will be rendered for can be defined in `variables.json`.
+
+For example:
 
 ```json
 {
-    "experiment_name": "Experiment_1"
+    "Qin_average_step": [20, 30]
 }
+
 ```
-If we have `steppedValueSet` and `enumeratedValueSet` variable in the `experiments.xml`  the corresponding template and data file will be as follows:
 
+Here, we want our `Qin_average_step` to be replaced by two different values `20` and `30`. As a result, two new files will be generated which contain the lines below.
+
+The first `experiments.xml` will contain:
 ```xml
-<steppedValueSet variable="Qin_average" first="40" step="10" last="200" /> 
-<enumeratedValueSet variable="Qin_randomizer"> <value value="0" />
+<steppedValueSet variable="Qin_average" first="30" step="20" last="200"/>
 ```
 
-```jinja2
-{% for variable, first, step, last in stepped_values %}
-<steppedValueSet variable="{{ Qin_average }}" first="{{ first }}" step="{{ step }}" last="{{ last }}" />
-{% endfor %}
-<enumeratedValueSet variable="{{ Qin_randomizer }}">
-<value value="{{ value }}" />
+The second `experiments.xml` will contain:
+```xml
+<steppedValueSet variable="Qin_average" first="30" step="20" last="200"/>
 ```
 
-```json
-"stepped_values": 
-[{
-  "variable": "Qin_average",
-  "first": 40,
-  "step": 10,
-  "last": 200
-  }
-],
-"Qin_randomizer":[0,1,2]
+In order to generate these files, we will need to follow one more step and run the Python code which will render the templates.
+
+### Step 3: Generate the experiments
+
+Once the variables are defined in the templates and the `variables.json` file, you can generate a new code by running the commands below.
+
+Create a new virtual environment:
+
+```shell
+python -m venv venv
+source ./venv/bin/activate
 ```
 
+Install the required dependencies:
 
+```shell
+python -m pip install -r requirements.txt
+```
+
+Generate the codes from the templates:
+
+```shell
+python generate_model.py
+```
+
+The output for the example in this documentation will be:
 
+```shell
+parsed variables: {"variables":{"Qin_average_step":[20,30]}}
 
+experiment:0  parameters:{'Qin_average_step': 20, 'config_id': '17b74c98-6ed3-4d01-8d05-d00fa77cfe57', 'experiment_name': 'experiment_0'}
+experiment:1  parameters:{'Qin_average_step': 30, 'config_id': 'dbea87fe-d6b1-421c-b622-e3d3e12698ac', 'experiment_name': 'experiment_1'}
+```
```

Comment:
To summarise:

1. Place all model files into the folder called `templates`
2. Add an extention `.jinja` to each file
3. Replace hard-coded values with a `{{variable}}`
4. Add `variables.json` file in the root folder in which values to be plugged instead of each variables are specified
5. Run `generate_model.py`
6. New rendered models will be in folder `experiments_n`, where `n` is the number of experiments equal to the number of variable combinations.